src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt

```diff
@@ -0,0 +1,57 @@
+package com.hibob.academy.resource
+
+import com.hibob.academy.entity.Owner
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.core.Response.Status
+import org.springframework.stereotype.Controller
+
+@Controller
+@Path("/api/owner")
+@Produces(MediaType.APPLICATION_JSON)
+class OwnerResource {
+
+    @GET
+    fun getALLOwners() : Response {
+        return Response.ok(listOf(
+            Owner(1, "Shahar", 1, 1, null,null),
+            Owner(2, "Alex", 1, 2, null,null )
+        ))
+            .build()
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun createOwner(owner: Owner): Response {
+
+        owner.name = owner.name ?: run {
+            if (owner.firstName.isNullOrEmpty() || owner.lastName.isNullOrEmpty()) {
+                return Response.status(Status.BAD_REQUEST).entity("Name is required").build()
+            } else {
+                "${owner.firstName} ${owner.lastName}"
+            }
+        }
+
+        owner.name?.let { fullName ->
+            val nameParts = fullName.split(" ")
+            owner.firstName = nameParts[0]
+            owner.lastName = nameParts.getOrNull(1)
```

Comment:
Something to consider. What happens if they don't send a last name. Also what happens if he has multiple spaces? So Jackie Van Halen for example, you would miss out on the rest of his name.

Reply:
If they don't provide a last name, I need to reject it, right? As for the second issue, I fixed it.